### PR TITLE
Documenting the git flow workflow and create sequence diagrams

### LINKED ISSUE #4043: [Documentation]: Documenting the GitFlow workflow and create sequence diagrams
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

6.0.0-rc.1

### Operating system

N/A

### What are you seeing?

Documenting the GitFlow workflow and create sequence diagrams

### What is expected?

The GitFlow workflow is documented

### Steps to Reproduce

n/a

### RepositoryFixture Test

_No response_

### Output log or link to your CI build (if appropriate).

_No response_

### PR DESCRIPTION
**Pull request by HHobeck**
## Description
Documenting the git flow workflow and create sequence diagrams

Close #4043

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by HHobeck** [comment]
Please review @asbjornu and/or @arturcic and merge to main

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 1]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
-[continuous deployment](/docs/reference/modes/continuous-deployment) mode for
-`develop` and [continuous delivery](/docs/reference/modes/continuous-delivery) mode
-for all other branches.
-
-This default configuration allows you to publish CI builds from develop to a CI
-MyGet feed, or another CI feed. Then all other branches are manually released
-then tagged. Read more about this at [version increments](/docs/reference/version-increments).
+These examples are illustrating the usage of the supported `GitFlow` workflow in GitVersion. 
+To enable this workflow, the build-in template [GitFlow/v1](/docs/workflows/GitFlow/v1.json) needs to be referenced in the configuration as following:
+```yaml
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:11]: Add a line-break before the code block.

```suggestion
To enable this workflow, the builtin template
[GitFlow/v1](/docs/workflows/GitFlow/v1.json) needs to be referenced in the
configuration as follows:

```yaml
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 2]
```diff
@@ -71,23 +89,17 @@ majors, then name your branch `support/<major>.x` (i.e `support/1.x`), to
...
 
 ## To Contribute
 
 See [contributing examples](/docs/learn/branching-strategies/contribute-examples).
 
 ### Source
 
-See `DocumentationSamples.GitFlowExample`. To update, modify then run test.
-Update <https://gist.github.com/JakeGinnivan/cf053d7f5d336ae9f7bb>
+See `DocumentationSamplesForGitFlow.cs`. To update, modify then run test.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:105]: Foot note links for the above footnote references.

```suggestion
See `DocumentationSamplesForGitFlow.cs`. To update, modify then run test.

[continuous-deployment]: /docs/reference/modes/continuous-deployment
[continuous-delivery]: /docs/reference/modes/continuous-delivery
[manual-deployment]: /docs/reference/modes/manual-deployment
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 3]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
+```yaml
+workflow: GitFlow/v1
+mode: ContinuousDelivery
+```
+
+Where 
+the [continuous deployment](/docs/reference/modes/continuous-deployment) mode for no branches, 
+the [continuous delivery](/docs/reference/modes/continuous-delivery) mode for `main`, `support` and `develop` branches and 
+the [manual deployment](/docs/reference/modes/manual-deployment) mode for `release`, `feature`, `hotfix` and `unknown` branches 
+are specified.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:20]: This section becomes easier to parse when the links are changed into footnote references.

```suggestion
Where the [continuous deployment][continuous-deployment] mode for no
branches, the [continuous delivery][continuous-delivery] mode for `main`,
`support` and `develop` branches and the [manual
deployment][manual-deployment] mode for `release`, `feature`, `hotfix` and
`unknown` branches are specified.
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 4]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
+
+This configuration allows you to publish CI (Continuous Integration) builds from `main`, `support` and `develop` branches to an artifact repository. 
+All other branches are manually published. Read more about this at [version increments](/docs/reference/version-increments).
+
+__Notice:__ The _continuous delivery_ mode has been used for the `main` and the `support` branch in this examples (specified as a fallback on the root configuration layer) 
+to illustrate how the version increments are applying. In production context the _continuous deployment_ mode might be a better option when e.g. 
+the releasing process is automated or the commits are tagged by the pipeline automatically.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:28]: This section can be wrapped in an alert.

I've also fixed a couple of typos in the text.

```suggestion
:::{.alert .alert-info}
The _continuous delivery_ mode has been used for the `main` and the
`support` branch in this examples (specified as a fallback on the root
configuration layer) to illustrate how the version increments are applied.
In production context the _continuous deployment_ mode might be a better
option when e.g. the release process is automated or the commits are tagged
by the pipeline automatically.
:::
```

**Comment by asbjornu** [comment]
I'm having a look at it now, @HHobeck. Looks like great work! A proper review will be done shortly.

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 5]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
 
 ## Feature Branches
 
-Feature branches will take the feature branch name and use that as the
-pre-release tag.
+Feature branches can be used in the `GitFlow` workflow to implement a feature in an isolated environement. Feature branches will take the feature branch name and use that 
+as the pre-release label. Will be created from `develop`, `release`, `main`, `support` or `hotfix` branches.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:32]: Wrap at 80 characters and fixed a typo (`environement` -> `environment`).

```suggestion
Feature branches can be used in the `GitFlow` workflow to implement a
feature in an isolated environment. Feature branches will take the feature
branch name and use that as the pre-release label. Feature branches will be
created from a `develop`, `release`, `main`, `support` or `hotfix` branch.
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 6]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
+![GitFlow](/docs/img/DocumentationSamplesForGitFlow\_FeatureFromMainBranch.png)
 
-You can see the difference on the feature branch itself, notice the version is
-the same before and after the commit on the feature branch? Only the metadata
-has changed. If you released the feature branch artifacts then tagged the
-commit, the following commit would increase to `-beta.2`.
+__Notice:__ After the feature branch is merged, the version on `main` is
+`2.0.0-5`. This is due to `main` running in _continuous delivery_
+mode. If you configured `main` to use _continuous deployment_ the version would
+be `2.0.0`.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:41]: Wrap in alert box and minor language tweak.

```suggestion
:::{.alert .alert-info}
After the feature branch is merged, the version on `main` is `2.0.0-5`.
This is due to `main` running in _continuous delivery_ mode. If `main` was
configured to use _continuous deployment_ the version would be `2.0.0`.
:::
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 7]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
-Because feature branches are most likely pushed to a fork, we are showing the
-pull request branch name which is created when you submit a pull request
+![GitFlow](/docs/img/DocumentationSamplesForGitFlow\_FeatureFromDevelopBranch.png)
 
-![GitFlow](/docs/img/09fdf46995b771f3164a_pull-request.png)
+__Notice:__ After the feature branch is merged, the version on `develop` is
+`1.3.0-alpha.3`. This is due to `develop` running in _continuous delivery_
+mode. If you configure `develop` to use _manual deployment_ the version would
+still be `1.3.0-alpha.1` and you would have to use pre-release tags to increment the
+`alpha.1`.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:51]: Wrap in alert box and minor language tweak.

```suggestion
:::{.alert .alert-info}
After the feature branch is merged, the version on `develop` is
`1.3.0-alpha.3`. This is due to `develop` running in _continuous delivery_
mode. If `develop` was configured to use _manual deployment_ the version
would still be `1.3.0-alpha.1` and you would have to use pre-release tags
to increment the pre-release label `alpha.1`.
:::
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 8]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
 
 ## Hotfix Branches
 
-Hotfix branches are used when you need to do a _patch_ release in GitFlow and
-are always created off `main`
+Hotfix branches are used when you need to do a _patch_ release in the `GitFlow` workflow and are always created from `main` branch.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:55]: Wrap at 80 characters.

```suggestion
Hotfix branches are used when you need to do a _patch_ release in the
`GitFlow` workflow and are always created from `main` branch.
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 9]
```diff
@@ -6,61 +6,79 @@ RedirectFrom:
...
+```
+
+Where 
+the [continuous deployment](/docs/reference/modes/continuous-deployment) mode for no branches, 
+the [continuous delivery](/docs/reference/modes/continuous-delivery) mode for `main`, `support` and `develop` branches and 
+the [manual deployment](/docs/reference/modes/manual-deployment) mode for `release`, `feature`, `hotfix` and `unknown` branches 
+are specified.
+
+This configuration allows you to publish CI (Continuous Integration) builds from `main`, `support` and `develop` branches to an artifact repository. 
+All other branches are manually published. Read more about this at [version increments](/docs/reference/version-increments).
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:23]: I prefer to wrap Markdown content at 80 characters, so the line-length is sensible, making the text easier to read in plain text. I thought that should have been enforced by the configured Markdown linter, but apparently not? 🤔 

```suggestion
This configuration allows you to publish CI (Continuous Integration) builds
from `main`, `support` and `develop` branches to an artifact repository.
All other branches are manually published. Read more about this at [version
increments](/docs/reference/version-increments).
```

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 10]
```diff
@@ -71,23 +89,17 @@ majors, then name your branch `support/<major>.x` (i.e `support/1.x`), to
...
 (i.e `support/1.3.x` or `support/1.3.0`)
 
-### Hotfix
+![GitFlow](/docs/img/DocumentationSamplesForGitFlow\_SupportBranch.png)
 
-Depending on what you name your support branch, you may or may not need a hotfix
+__Notice:__ Depending on what you name your support branch, you may or may not need a hotfix
 branch. Naming it `support/1.x` will automatically bump the patch, if you name
 it `support/1.3.0` then the version in branch name rule will kick in and the
 patch _will not_ automatically bump, meaning you have to use hotfix branches.
```
Comment (asbjornu) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:97]: Wrap in alert box and add a newline after.

```suggestion
:::{.alert .alert-info}
Depending on what you name your support branch, you may or may not need a
hotfix branch. Naming it `support/1.x` will automatically bump the patch,
if you name it `support/1.3.0` then the version in branch name rule will
kick in and the patch _will not_ automatically bump, meaning you have to
use hotfix branches.
:::

```

**Review by asbjornu — APPROVED** [review]

**Pull request merged**